x-pack/plugins/synthetics/public/apps/synthetics/state/monitor_management/monitor_list.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { takeLeading } from 'redux-saga/effects';
+import {
+  FetchMonitorManagementListQueryArgs,
+  MonitorManagementListResult,
+  MonitorManagementListResultCodec,
+} from '../../../../../common/runtime_types';
+import { SyntheticsAppState } from '../root_reducer';
+import { fetchEffectFactory } from '../utils/fetch_effect';
+import { apiService } from '../../../../utils/api_service';
+import { API_URLS } from '../../../../../common/constants';
+
+export const monitorListSlice = createSlice({
+  name: 'monitorList',
+  initialState: {
+    data: {} as MonitorManagementListResult,
+    loading: false,
+  },
+  reducers: {
+    fetchMonitorListAction: (state, action) => {
+      return {
+        ...state,
+        loading: true,
+      };
+    },
+    fetchMonitorListActionSuccess: (state, action: PayloadAction<MonitorManagementListResult>) => {
+      return {
+        ...state,
+        loading: false,
+        data: action.payload,
+      };
+    },
+    fetchMonitorListActionFail: (state, action) => {
+      return {
+        ...state,
+        loading: false,
+        data: action.payload,
```

Comment:
With `createSlice`, it's easy to do mistakes as types have to be defined implicitly. E.g. here `action.payload` is an IHttpError assigned to `data` of type MonitorManagementListResult without complains. 

Redux toolkit [recommends](https://redux-toolkit.js.org/api/createReducer#usage-with-the-builder-callback-notation) the builder syntax for reducers for better TypeScript. Another option could be to use the `extraReducers` option of `createSlice` which also accepts the builder syntax, [see](https://redux-toolkit.js.org/api/createslice#examples).

Excerpt from Redux Toolkit:
> The recommended way of using `createReducer` is the builder callback notation, as it works best with TypeScript and most IDEs.

We can pick one and stick to it. 

Another simplification we can do is to mutate the state e.g.
Instead of 
``` typescript
      return {
        ...state,
        loading: false,
        data: action.payload
      }
```

We can just do
``` typescript
     state.loading = false;
     state.data = action.payload;
```
Because Redux Toolkit already [takes care](https://redux-toolkit.js.org/api/createReducer#direct-state-mutation). Excerpt:

>To make things easier, createReducer uses [immer](https://github.com/mweststrate/immer) to let you write reducers as if they were mutating the state directly. In reality, the reducer receives a proxy state that translates all mutations into equivalent copy operations.